workshops/2021-16-02_intro_to_github/TOG_git_tutorial.Rmd

```diff
@@ -0,0 +1,396 @@
+
+Paste your key into the “Key” field.
+
+Click Add SSH key.
+
+Confirm the action by entering your GitHub password.
+
+
+## part 3: Create, Clone, Stage, Commit, Push, Pull!
+
+Now, we're ready to start using Git.
+
+The six verbs in the title will likely encompass 90% of everything you will do with Git. We will now walk through each one to give a brief explanation and demo. 
+
+Note that everything that can be done using RStudio (or any other Git Client), can also be done via the shell. Here we provide instructions for RStudio environment for a more guided experience. 
+
+If you're interested, check out [this post](http://dont-be-afraid-to-commit.readthedocs.io/en/latest/git/commandlinegit.html) on using Git in the shell. It is definitely a worthwhile learning experience if you want to dive deeper into the Git world.
+
+### Overview
+
+A repository is simply a directory for your project. It is no different from any other directories on your laptop, except that Git knows about it and keeps a complete historical record of every change that has been "committed". 
+
+When a repository is stored on GitHub, it is called the "remote" repository. You can then "clone" this repository onto your laptop. The copy on your laptop is then called the "local" repository. "Pushing" and "pulling" are operations for keeping the local and remote repositories in sync with each other. 
+
+For example, You can modify codes in your "local" repository and then, when ready, "push" your changes to the "remote" repository on GitHub. Your group members can then "pull"" the most up-to-date version of the "remote" repository from GitHub into their "local" repositories and continue to work on the project. 
+
+Don't worry if you don't understand every detail of the workflow just yet. Just get the overall picture for now. You'll gain a much deeper understanding once we start trying things out for real, starting now!
+
+### Create 
+
+**Update 14 January 2021** *Skip the create step for this deliverable; instead use the private repo that was created for you (named `zz_Your-Name_STAT540_2021W2`) - refer back to these instructions in your future work when you'd like to create a new repo.* 
+
+To create a repository, go to [GitHub](https://github.com/). If you're logged in, you should see a "New repository" button on your home screen. Click it. 
+
+On the "Create a new repository" page, make sure your GitHub account has been selected as the owner. Give your repository a name. I gave mine "testing-repository". Enter a description if you like. Choose public for now. Don't worry about initializing the README file; we will add that in later manually. 
+
+Click "Create repository". If you see this (screenshot), things are going well. This is the web page of your new repository on Github. Using the GitHub website, you will be able to browse your files directly inside of an internet browser. 
+
+Congratulations!
+
+![Repository created screenshot](create_repository_success.png)
```

Comment:
Perhaps knit to a markdown (the default for github_document) so it can be viewed directly on github. It could be useful to go over knitting and pushing changes to github as well since a big plus of github is sharing analysis protocols with your lab